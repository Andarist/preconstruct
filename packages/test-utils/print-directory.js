// @flow
// wanted to make a tree view of a directory with the contents of the files
// for testing but gonna finish it later
// based on https://github.com/egoist/tre/blob/master/src/index.js
import * as fs from "fs-extra";
import path from "path";

const CHARS = {
  continue: "├──",
  end: "└──",
  line: "│"
};

function repeat(char, count) {
  let res = "";
  while (count--) {
    res += char;
  }
  return res;
}

async function getData(
  directory: string,
  depth: number,
  items: Array<{ type: "file" | "directory", depth: number, name: string }>
) {
  let dir = await fs.readdir(directory);

  // this should probably happen in parallel but not sure how to make ordering good

  for (let item of dir) {
    let joined = path.join(directory, item);
    let stats = await fs.stat(joined);
    if (stats.isDirectory()) {
      items.push();
      await getData(joined, depth + 1, items);
    } else if (stats.isFile()) {
      // items.push({});
    } else {
      throw new Error("unknown thing");
    }
  }

  return items;
}

export default async function printDirectory(directory: string) {
  let data = [];
  await getData(directory, 0, data);

  return (
    `.\n` +
    data
      .map((item, index) => {
        const isLast = index === data.length - 1;
        const next = !isLast && data[index + 1];
        const char =
          next && next.depth === item.depth ? CHARS.continue : CHARS.end;
        const linePrefix = item.depth > 0 && !isLast ? CHARS.line : "";
        const prefix = `${linePrefix}${repeat(
          " ",
          item.depth === 0 ? 0 : 3 + (item.depth - 1) * 4 + (linePrefix ? 0 : 1)
        )}${char}`;

        return `${prefix} ${item.name}`;
      })
      .join("\n")
  );
}
