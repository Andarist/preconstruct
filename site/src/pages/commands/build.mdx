# `build`

```bash
preconstruct build
```

`preconstruct build` creates the bundles for your project.
