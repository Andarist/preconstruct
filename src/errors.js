// @flow
"use strict";

class FatalError extends Error {}

module.exports = { FatalError };
