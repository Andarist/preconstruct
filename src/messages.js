// @flow

export let errors = {
  noEntryPoint:
    "no entrypoint was provided, please create a file at src/index.js",
  deniedWriteMainField:
    "changing the main field is required for preconstruct to build",
  invalidModuleField: "module field is invalid",
  invalidMainField: "main field is invalid",
  invalidUmdMainField: "umd:main field is invalid",
  umdNameNotSpecified:
    "the umd:main field is specified but a umdName option is not specified. please add it to the preconstruct field in your package.json"
};

export let confirms = {
  writeMainField:
    "preconstruct is going to change the main field in your package.json, are you okay with that?",
  writeModuleField:
    "would you like to generate module builds? this will write to the module field in your package.json",
  fixModuleField: "would you like to fix the module field?",
  writeUmdBuilds: "would you like to generate UMD builds?"
};

export let inputs = {
  getUmdName: "what should the name used for UMD bundles be?"
};

export let infos = {
  validMainField: "main field is valid",
  validModuleField: "module field is valid",
  validUmdMainField: "umd:main field is valid",
  validEntrypoint: "a valid entry point exists."
};

export let successes = {
  validPackage: "package is valid!"
};
