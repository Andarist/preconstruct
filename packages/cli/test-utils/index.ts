import path from "path";
import * as fs from "fs-extra";
import fastGlob from "fast-glob";
import fixturez from "fixturez";
import spawn from "spawndamnit";
import outdent from "outdent";
import crypto from "crypto";
// import profiler from "v8-profiler-next";
import chalk from "chalk";

let f = fixturez(__dirname);

export const js = outdent;
export const ts = outdent;
export const tsx = outdent;

chalk.level = 0;

console.error = jest.fn();
console.log = jest.fn();

export let logMock = {
  log: (console.log as any) as jest.MockInstance<void, any>,
  error: (console.error as any) as jest.MockInstance<void, any>,
};

afterEach(() => {
  logMock.log.mockReset();
  logMock.error.mockReset();
});

import init from "../src/init";
import { confirms } from "../src/messages";

let mockedConfirms = confirms as jest.Mocked<typeof confirms>;

export async function initBasic(directory: string) {
  mockedConfirms.writeMainField.mockReturnValue(Promise.resolve(true));
  mockedConfirms.writeModuleField.mockReturnValue(Promise.resolve(true));

  await init(directory);
  mockedConfirms.writeMainField.mockReset();
  mockedConfirms.writeModuleField.mockReset();
}

// export function profile(name: string) {
//   profiler.startProfiling(name, true);

//   return () => {
//     let profile = profiler.stopProfiling();

//     new Promise<void>((resolve) =>
//       profile
//         .export()
//         .pipe(fs.createWriteStream(name + ".cpuprofile"))
//         .on("finish", function () {
//           profile.delete();
//           resolve();
//         })
//     );
//   };
// }

function getPkgPath(tmpPath: string) {
  return path.join(tmpPath, "package.json");
}

export async function getPkg(
  filepath: string
): Promise<{ [key: string]: any }> {
  return JSON.parse(
    await fs.readFile(path.join(filepath, "package.json"), "utf-8")
  );
}

export async function modifyPkg(
  tmpPath: string,
  cb: (pkgJson: { [key: string]: any }) => unknown
) {
  let json = await getPkg(tmpPath);
  await cb(json);

  let pkgPath = getPkgPath(tmpPath);
  await fs.writeFile(pkgPath, JSON.stringify(json, null, 2));
}

export let createPackageCheckTestCreator = (
  doResult: (tmpPath: string) => Promise<void>
) => {
  let createTestCreator = (
    testFn: (testName: string, func: () => Promise<void>) => void
  ) => async (
    testName: string,
    entrypoints: { [key: string]: any },
    cb: (
      doThing: () => Promise<{ [key: string]: any }>,
      stuff: { dir: string }
    ) => Promise<void>
  ) => {
    testFn(testName, async () => {
      let tmpPath = f.copy("template-simple-package");
      let things = Object.keys(entrypoints);
      await Promise.all(
        things.map(async (entrypointPath) => {
          let content = entrypoints[entrypointPath];
          let filepath = path.join(tmpPath, entrypointPath, "package.json");
          await fs.ensureFile(filepath);
          await fs.writeFile(filepath, JSON.stringify(content, null, 2));
        })
      );

      await cb(
        async () => {
          await doResult(tmpPath);

          let newThings: Record<string, any> = {};

          await Promise.all(
            things.map(async (entrypointPath) => {
              newThings[entrypointPath] = JSON.parse(
                await fs.readFile(
                  path.join(tmpPath, entrypointPath, "package.json"),
                  "utf8"
                )
              );
            })
          );
          return newThings;
        },
        { dir: tmpPath }
      );
    });
  };
  let testFn = createTestCreator(test);
  // @ts-ignore
  testFn.only = createTestCreator(test.only);
  // @ts-ignore
  testFn.skip = createTestCreator(test.skip);
  return testFn as typeof testFn & { only: typeof testFn; skip: typeof testFn };
};

export async function snapshotDistFiles(tmpPath: string) {
  let distPath = path.join(tmpPath, "dist");
  let distFiles;
  try {
    distFiles = await fs.readdir(distPath);
  } catch (err) {
    if (err.code === "ENOENT") {
      throw new Error(distPath + " does not exist");
    }
    throw err;
  }

  await Promise.all(
    distFiles.map(async (x) => {
      expect(
        await fs.readFile(path.join(distPath, x), "utf-8")
      ).toMatchSnapshot(x);
    })
  );
}

function hash(content: string) {
  return crypto.createHash("md5").update(content).digest("hex");
}

export let stripHashes = async (chunkName: string) => {
  let transformer = (pathname: string, content: string) => {
    return pathname.replace(new RegExp(`${chunkName}-[^\\.]+`, "g"), () => {
      return `chunk-this-is-not-the-real-hash-${hash(content)}`;
    });
  };
  return {
    transformPath: transformer,
    transformContent: (content: string) => {
      return content.replace(new RegExp(`${chunkName}-[^\\.]+`, "g"), () => {
        return "chunk-some-hash";
      });
    },
  };
};

export async function snapshotDirectory(
  tmpPath: string,
  {
    files = "js",
    filterPath = (x) => true,
    transformPath = (x) => x,
    transformContent = (x) => x,
  }: {
    files?: "all" | "js";
    filterPath?: (path: string) => boolean;
    transformPath?: (path: string, contents: string) => string;
    transformContent?: (content: string) => string;
  } = {}
) {
  let paths = await fastGlob(
    [`**/${files === "js" ? "*.js" : "*"}`, "!node_modules/**", "!yarn.lock"],
    {
      cwd: tmpPath,
    }
  );

  await Promise.all(
    paths
      .filter((fp) => filterPath(fp))
      .map(async (x) => {
        let content = transformContent(
          await fs.readFile(path.join(tmpPath, x), "utf-8")
        );
        if (x.endsWith(".json") && !x.endsWith("tsconfig.json")) {
          content = JSON.parse(content);
        }
        expect(content).toMatchSnapshot(transformPath(x, content));
      })
  );
}

export async function install(tmpPath: string) {
  await spawn("yarn", ["install"], { cwd: tmpPath });
}

export const repoNodeModules = path.resolve(
  __dirname,
  "..",
  "..",
  "..",
  "node_modules"
);

export async function testdir(dir: {
  [key: string]: string | { kind: "symlink"; path: string };
}) {
  const temp = f.temp();
  await Promise.all(
    Object.keys(dir).map(async (filename) => {
      const output = dir[filename];
      const fullPath = path.join(temp, filename);
      if (typeof output === "string") {
        await fs.outputFile(fullPath, dir[filename]);
      } else {
        const dir = path.dirname(fullPath);
        await fs.ensureDir(dir);
        await fs.symlink(path.resolve(temp, output.path), fullPath);
      }
    })
  );
  return temp;
}

expect.addSnapshotSerializer({
  print(val: Record<string, string>, serialize, indent) {
    const contentsByFilename: Record<string, string[]> = {};
    Object.entries(val).forEach(([filename, contents]) => {
      if (contentsByFilename[contents] === undefined) {
        contentsByFilename[contents] = [];
      }
      contentsByFilename[contents].push(filename);
    });
    return Object.entries(contentsByFilename)
      .map(([contents, filenames]) => {
        return `${filenames.join(", ")} -------------\n${contents}`;
      })
      .join("\n");
  },
  test(val) {
    return val && val[dirPrintingSymbol];
  },
});

const dirPrintingSymbol = Symbol("dir printing symbol");

export async function getDist(dir: string) {
  return getFiles(dir, ["dist/**"]);
}

export async function getFiles(dir: string, glob: string[] = ["**"]) {
  const files = await fastGlob(glob, { cwd: dir });
  const filesObj: Record<string, string> = {
    [dirPrintingSymbol]: true,
  };
  await Promise.all(
    files.map(async (filename) => {
      filesObj[filename] = await fs.readFile(path.join(dir, filename), "utf8");
    })
  );
  let newObj: Record<string, string> = { [dirPrintingSymbol]: true };
  files.sort().forEach((filename) => {
    newObj[filename] = filesObj[filename];
  });
  return newObj;
}

export const basicPkgJson = (options?: { module?: boolean }) => {
  return JSON.stringify({
    name: "pkg",
    main: "dist/pkg.cjs.js",
    module: options?.module ? "dist/pkg.esm.js" : undefined,
  });
};
