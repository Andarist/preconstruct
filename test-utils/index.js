// @flow
import path from "path";
import * as fs from "fs-extra";
require("chalk").enabled = false;
// $FlowFixMe
console.error = jest.fn();
// $FlowFixMe
console.log = jest.fn();

export let logMock = {
  log: ((console.log: any): JestMockFn<any, void>),
  error: ((console.error: any): JestMockFn<any, void>)
};

afterEach(() => {
  logMock.log.mockReset();
  logMock.error.mockReset();
});

import init from "../src/init";
import { confirms } from "../src/messages";

export async function initBasic(directory: string) {
  confirms.writeMainField.mockReturnValue(true);
  confirms.writeModuleField.mockReturnValue(true);
  confirms.writeUmdBuilds.mockReturnValue(false);

  await init(directory);
  confirms.writeMainField.mockReset();
  confirms.writeModuleField.mockReset();
  confirms.writeUmdBuilds.mockReset();
}

function getPkgPath(tmpPath: string) {
  return path.join(tmpPath, "package.json");
}

export async function getPkg(filepath: string): Object {
  return JSON.parse(
    await fs.readFile(path.join(filepath, "package.json"), "utf-8")
  );
}

export async function modifyPkg(tmpPath: string, cb: Object => mixed) {
  let json = await getPkg(tmpPath);
  await cb(json);

  let pkgPath = getPkgPath(tmpPath);
  await fs.writeFile(pkgPath, JSON.stringify(json, null, 2));
}
