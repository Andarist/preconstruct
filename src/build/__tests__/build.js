// @flow
import build from "../";
import fixturez from "fixturez";
import path from "path";
import * as fs from "fs-extra";
import { initBasic } from "../../../test-utils";
import { confirms } from "../../messages";
import { FatalError } from "../../errors";

const f = fixturez(__dirname);

async function snapshotDistFiles(tmpPath: string) {
  let distPath = path.join(tmpPath, "dist");
  let distFiles;
  try {
    distFiles = await fs.readdir(distPath);
  } catch (err) {
    if (err.code === "ENOENT") {
      throw new Error("no dist directory exists");
    }
    throw err;
  }

  await Promise.all(
    distFiles.map(async x => {
      expect(
        await fs.readFile(path.join(distPath, x), "utf-8")
      ).toMatchSnapshot(x);
    })
  );
}

jest.mock("install-packages");

let unsafeRequire = require;

test("basic", async () => {
  let tmpPath = f.copy("valid-package");

  await build(tmpPath);

  await snapshotDistFiles(tmpPath);

  expect(unsafeRequire(tmpPath).default).toBe("something");
});

test("monorepo", async () => {
  let tmpPath = f.copy("monorepo");
  await initBasic(tmpPath);
  await build(tmpPath);
  let counter = 1;
  for (let pkg of ["package-one", "package-two"]) {
    let pkgPath = path.join(tmpPath, "packages", pkg);
    await snapshotDistFiles(pkgPath);

    expect(unsafeRequire(pkgPath).default).toBe(counter++);
  }
});

test("no module", async () => {
  let tmpPath = f.copy("no-module");

  await build(tmpPath);

  await snapshotDistFiles(tmpPath);

  expect(unsafeRequire(tmpPath).default).toBe(
    "this does not have a module build"
  );
});

test.skip("uses obj spread", async () => {
  let tmpPath = f.copy("use-obj-spread");
  confirms.shouldInstallObjectAssign.mockReturnValue(Promise.resolve(false));

  try {
    await build(tmpPath);
  } catch (err) {
    expect(err).toBeInstanceOf(FatalError);
    expect(err.message).toBe(
      "object-assign should be in dependencies of use-object-spread"
    );
  }
});

test("object-assign installed", async () => {
  let tmpPath = f.copy("object-assign-installed");

  await build(tmpPath);

  await snapshotDistFiles(tmpPath);
});

test("clears dist folder", async () => {
  let tmpPath = f.copy("already-has-things-in-dist");

  await build(tmpPath);

  await snapshotDistFiles(tmpPath);
});

test("flow", async () => {
  let tmpPath = f.copy("flow");

  await build(tmpPath);

  await snapshotDistFiles(tmpPath);
});

test("flow", async () => {
  let tmpPath = f.copy("flow-export-default");

  await build(tmpPath);

  await snapshotDistFiles(tmpPath);
});

test("prod checks", async () => {
  let tmpPath = f.copy("prod-checks");

  await build(tmpPath);

  await snapshotDistFiles(tmpPath);
});

// TODO: make it faster so this isn't required
jest.setTimeout(20000);

test("browser", async () => {
  let tmpPath = f.copy("browser");

  confirms.addBrowserField.mockReturnValue(Promise.resolve(true));

  await build(tmpPath);

  await snapshotDistFiles(tmpPath);
});
