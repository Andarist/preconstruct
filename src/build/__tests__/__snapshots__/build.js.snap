// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic: valid-package.cjs.dev.js 1`] = `
"'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var index = \\"something\\";

exports.default = index;
"
`;

exports[`basic: valid-package.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./valid-package.cjs.prod.js\\");
} else {
  module.exports = require(\\"./valid-package.cjs.dev.js\\");
}"
`;

exports[`basic: valid-package.cjs.prod.js 1`] = `
"\\"use strict\\";
Object.defineProperty(exports, \\"__esModule\\", { value: !0 });
var index = \\"something\\";
exports.default = index;
"
`;

exports[`basic: valid-package.esm.js 1`] = `
"var index = \\"something\\";

export default index;
"
`;

exports[`basic: valid-package.umd.min.js 1`] = `
"!function(e,t){\\"object\\"==typeof exports&&\\"undefined\\"!=typeof module?module.exports=t():\\"function\\"==typeof define&&define.amd?define(t):e.validPackage=t()}(this,function(){\\"use strict\\";return\\"something\\"});
//# sourceMappingURL=valid-package.umd.min.js.map
"
`;

exports[`basic: valid-package.umd.min.js.map 1`] = `"{\\"version\\":3,\\"file\\":\\"valid-package.umd.min.js\\",\\"sources\\":[\\"../src/index.js\\"],\\"sourcesContent\\":[\\"export default \\\\\\"something\\\\\\";\\\\n\\"],\\"names\\":[],\\"mappings\\":\\"8LAAe\\"}"`;

exports[`clears dist folder: already-has-things-in-dist.cjs.dev.js 1`] = `
"'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var index = \\"something\\";

exports.default = index;
"
`;

exports[`clears dist folder: already-has-things-in-dist.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./already-has-things-in-dist.cjs.prod.js\\");
} else {
  module.exports = require(\\"./already-has-things-in-dist.cjs.dev.js\\");
}"
`;

exports[`clears dist folder: already-has-things-in-dist.cjs.prod.js 1`] = `
"\\"use strict\\";
Object.defineProperty(exports, \\"__esModule\\", { value: !0 });
var index = \\"something\\";
exports.default = index;
"
`;

exports[`clears dist folder: object-assign-installed.cjs.dev.js 1`] = `
"'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var _objectAssign = _interopDefault(require('object-assign'));

var something = _objectAssign({}, {
  something: true
});

exports.something = something;
"
`;

exports[`clears dist folder: object-assign-installed.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./object-assign-installed.cjs.prod.js\\");
} else {
  module.exports = require(\\"./object-assign-installed.cjs.dev.js\\");
}"
`;

exports[`clears dist folder: object-assign-installed.cjs.prod.js 1`] = `
"\\"use strict\\";
function _interopDefault(ex) {
  return ex && \\"object\\" == typeof ex && \\"default\\" in ex ? ex.default : ex;
}
Object.defineProperty(exports, \\"__esModule\\", { value: !0 });
var _objectAssign = _interopDefault(require(\\"object-assign\\")),
  something = _objectAssign({}, { something: !0 });
exports.something = something;
"
`;

exports[`flow: flow.cjs.dev.js 1`] = `
"'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var a = \\"wow\\";

function doSomething(arg) {
  return \\"something\\";
}

exports.doSomething = doSomething;
exports.something = a;
"
`;

exports[`flow: flow.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./flow.cjs.prod.js\\");
} else {
  module.exports = require(\\"./flow.cjs.dev.js\\");
}"
`;

exports[`flow: flow.cjs.js.flow 1`] = `
"// @flow
export * from \\"../src/index.js\\";
"
`;

exports[`flow: flow.cjs.prod.js 1`] = `
"\\"use strict\\";
Object.defineProperty(exports, \\"__esModule\\", { value: !0 });
var a = \\"wow\\";
function doSomething(arg) {
  return \\"something\\";
}
(exports.doSomething = doSomething), (exports.something = a);
"
`;

exports[`flow: flow.esm.js 1`] = `
"var a = \\"wow\\";

function doSomething(arg) {
  return \\"something\\";
}

export { doSomething, a as something };
"
`;

exports[`monorepo: package-one.cjs.dev.js 1`] = `
"'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var index = 1;

exports.default = index;
"
`;

exports[`monorepo: package-one.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./package-one.cjs.prod.js\\");
} else {
  module.exports = require(\\"./package-one.cjs.dev.js\\");
}"
`;

exports[`monorepo: package-one.cjs.prod.js 1`] = `
"\\"use strict\\";
Object.defineProperty(exports, \\"__esModule\\", { value: !0 });
var index = 1;
exports.default = index;
"
`;

exports[`monorepo: package-one.esm.js 1`] = `
"var index = 1;

export default index;
"
`;

exports[`monorepo: package-two.cjs.dev.js 1`] = `
"'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var index = 2;

exports.default = index;
"
`;

exports[`monorepo: package-two.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./package-two.cjs.prod.js\\");
} else {
  module.exports = require(\\"./package-two.cjs.dev.js\\");
}"
`;

exports[`monorepo: package-two.cjs.prod.js 1`] = `
"\\"use strict\\";
Object.defineProperty(exports, \\"__esModule\\", { value: !0 });
var index = 2;
exports.default = index;
"
`;

exports[`monorepo: package-two.esm.js 1`] = `
"var index = 2;

export default index;
"
`;

exports[`no module: no-module.cjs.dev.js 1`] = `
"'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var index = \\"this does not have a module build\\";

exports.default = index;
"
`;

exports[`no module: no-module.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./no-module.cjs.prod.js\\");
} else {
  module.exports = require(\\"./no-module.cjs.dev.js\\");
}"
`;

exports[`no module: no-module.cjs.prod.js 1`] = `
"\\"use strict\\";
Object.defineProperty(exports, \\"__esModule\\", { value: !0 });
var index = \\"this does not have a module build\\";
exports.default = index;
"
`;

exports[`prod checks: prod-checks.cjs.dev.js 1`] = `
"'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

function thing() {
  if (process.env.NODE_ENV !== \\"production\\") {
    return \\"not prod\\";
  }

  return \\"prod\\";
}

exports.thing = thing;
"
`;

exports[`prod checks: prod-checks.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./prod-checks.cjs.prod.js\\");
} else {
  module.exports = require(\\"./prod-checks.cjs.dev.js\\");
}"
`;

exports[`prod checks: prod-checks.cjs.prod.js 1`] = `
"\\"use strict\\";
function thing() {
  return \\"prod\\";
}
Object.defineProperty(exports, \\"__esModule\\", { value: !0 }),
  (exports.thing = thing);
"
`;
